fgd_cc_contacts/fgd_cc_contacts.py

```diff
@@ -0,0 +1,167 @@
+import random
+import time
+
+from core_data_modules.cleaners import Codes
+from core_data_modules.traced_data import Metadata, TracedData
+from core_data_modules.traced_data.io import TracedDataJsonIO, TracedDataCSVIO
+from core_data_modules.util import IOUtils, PhoneNumberUuidTable
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Cleans the wt surveys and exports variables to Coda for "
+                                                 "manual verification and coding")
+    parser.add_argument("user", help="User launching this program, for use by TracedData Metadata")
+    parser.add_argument("phone_uuid_table_path", metavar="phone-uuid-table-path",
+                        help="JSON file containing an existing phone number <-> UUID lookup table. ")
+    parser.add_argument("fgd_cc_input_path", metavar="fgd-cc-input-path",
+                        help="Path to input file, containing a list of serialized TracedData objects as JSON")
+    parser.add_argument("demog_surveys_input_path", metavar="demog-surveys-input-path",
+                        help="Path to input file, containing a list of serialized TracedData objects as JSON")
+    parser.add_argument("json_output_path", metavar="json-output-path",
+                        help="Path to a JSON file to write processed TracedData messages to")
+    parser.add_argument("fgd_csv_output_path", metavar="fgd-csv-output-path",
+                        help="Path to a CSV file to write 160 call centre contacts to")
+    parser.add_argument("cc_csv_output_path", metavar="cc-csv-output-path",
+                        help="Path to a CSV file to write 40 focus group discussion contacts to")
+
+    args = parser.parse_args()
+    user = args.user
+    phone_uuid_table_path = args.phone_uuid_table_path
+    fgd_cc_input_path = args.fgd_cc_input_path
+    demog_surveys_input_path = args.demog_surveys_input_path
+    json_output_path = args.json_output_path
+    fgd_csv_output_path = args.fgd_csv_output_path
+    cc_csv_output_path = args.cc_csv_output_path
+
+    # Load phone uuid table
+    with open(phone_uuid_table_path, "r") as f:
+        phone_uuids = PhoneNumberUuidTable.load(f)
+
+    # Load FGD/CC survey responses
+    with open(fgd_cc_input_path, "r") as f:
+        fgd_cc_data = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    # Load coded demog surveys
+    with open(demog_surveys_input_path, "r") as f:
+        surveys = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    # Filter out people who haven't answered the fgd_cc consent question
+    fgd_cc_consent_key = "Response_1 (Category) - wt_fgd_cc"
+    fgd_cc_data = [td for td in fgd_cc_data if fgd_cc_consent_key in td]
+
+    # Apply the demog surveys to the fgd_cc data
+    TracedData.update_iterable(user, "avf_phone_id", fgd_cc_data, surveys, "surveys")
+
+    # Annotate fgd_cc_data with whether or not the respondent's is from Mogadishu
+    mogadishu_districts = [
```

Comment:
These feel like they should be in a location that makes them reusable

Reply:
There is a MOGADISHU_DISTRICTS constant in Core, but it uses slightly different district names, so not going to import that here